### PR TITLE
Bug 1251306 — Fallback read-only buffer bookmarks

### PR DESCRIPTION
**Pull request by rnewman**
This is on top of the main bookmarks work in #1551.

In short: we extract out some factory stuff, and then we implement a split-brain factory that shows desktop bookmarks from the buffer and mobile bookmarks from local.

I also made some improvements around swipe-to-delete that will be folded into earlier commits, and also added the flag that can be used to disable merging, so this becomes easy to test on clean profiles.

### CODE REVIEW

**Comment by rnewman** [comment]
@sleroux, @fluffyemily, @ncalexan, et al: I think this is ready to go. No telemetry yet (Bug 1251689).

File: Storage/SQL/SQLiteBookmarksModel.swift [hunk 1]
```diff
@@ -0,0 +1,832 @@
...
+                bookmark.favicon = Favicon(url: faviconURL,
+                                           date: NSDate(timeIntervalSince1970: date),
+                                           type: type)
+        }
+    }
+
+    private class func livemarkFactory(row: SDRow) -> BookmarkItem {
+        let id = row["id"] as! Int
+        let guid = row["guid"] as! String
+        let url = row["siteUri"] as! String
```
Comment (fluffyemily) [inline Storage/SQL/SQLiteBookmarksModel.swift]: referencing `siteUri` here causes a crash. There is no `siteUri` in the row columnNames. Changing this to `feedUri` seems to work.

Comment (fluffyemily) [inline Storage/SQL/SQLiteBookmarksModel.swift]: It is also possible that the solution is to add `siteUri` to the list of fetched properties when querying bookmarks and their children (i.e. so they are there when this factory is executed)

Comment (rnewman) [inline Storage/SQL/SQLiteBookmarksModel.swift]: Yeah, we simply omitted the column from the query. It should be there.

**Comment by rnewman** [comment]
Rebased and merged. Thanks, all.

**Pull request closed without merging**